src/main/java/htsjdk/variant/variantcontext/GenotypeLikelihoods.java

```diff
@@ -492,10 +501,14 @@ public static GenotypeLikelihoodsAllelePair getAllelePair(final int PLindex) {
      * Initialize cache of allele anyploid indices
      * If initialized multiple times with the same ploidy, the alternate alleles from the last initialization will be used
      *
+     * This method is no longer necessary for this class, and in general should be avoided.  However it remains to avoid
+     * breaking the API
```

Comment:
I think those tests should just be dropped.  `testGetAlleles` does all the necessary testing, since we aren't building a cache separately from the `getAlleles` method anymore.